paymentapi/src/main/java/fi/hel/verkkokauppa/payment/service/PaymentMethodService.java

```diff
@@ -0,0 +1,233 @@
+package fi.hel.verkkokauppa.payment.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import fi.hel.verkkokauppa.common.constants.OrderType;
+import fi.hel.verkkokauppa.common.constants.PaymentType;
+import fi.hel.verkkokauppa.common.error.CommonApiException;
+import fi.hel.verkkokauppa.common.error.Error;
+import fi.hel.verkkokauppa.payment.api.data.GetPaymentMethodListRequest;
+import fi.hel.verkkokauppa.payment.api.data.OrderDto;
+import fi.hel.verkkokauppa.payment.api.data.PaymentMethodDto;
+import fi.hel.verkkokauppa.payment.api.data.PaymentMethodFilter;
+import fi.hel.verkkokauppa.payment.constant.GatewayEnum;
+import fi.hel.verkkokauppa.payment.repository.PaymentMethodRepository;
+import fi.hel.verkkokauppa.payment.util.CurrencyUtil;
+import fi.hel.verkkokauppa.payment.logic.fetcher.PaymentMethodListFetcher;
+import org.helsinki.vismapay.model.paymentmethods.PaymentMethod;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+@Service
+public class PaymentMethodService {
```

Comment:
I added methods that perform other operations than reading (create, update and delete), so I thought it would be good to rename it now that the service does not only do read operations. Optionally I could do a new Service class that has the create, update and delete operations and change this class back to ListService as it would then only have read operations. What do you think?

Reply:
That is good for now! Good job